### PR TITLE
Refactor find.turn.point.R and find.tol.time.R and recover.weaker.R

### LINKED ISSUE #74: refactor recover.weaker.R

### LINKED ISSUE #77: Refactor find.tol.time
- [ ] reformat file
- [ ] refactor operations to be done on a [tibble](http://statseducation.com/Introduction-to-R/modules/getting%20data/tibbles/)
- [ ] search for functions which can be re-used (compute breaks or so)
- [ ] check usage in `adjust.time` and `feature.align` functions and remove the not required return values

### LINKED ISSUE #90: refactor find.turnpoints
Continuation see #94

### PR DESCRIPTION
**Pull request by wverastegui**
Refactored  find.tol.time.R and find.turn.point.R. The pastecs library has been added to use the Turnpoint function in find.turn.point.R, that simplifies the code. Added pastecs to dependencies.

Closes #90 
Closes #77 

This PR now also includes the changes from https://github.com/RECETOX/recetox-aplcms/pull/92 to fix the overarching test cases, as fixing the things for this PR would require changes in other functions, so I decided to append all changes from https://github.com/RECETOX/recetox-aplcms/pull/92 here.

- Extracted all functions performing substeps of the computation
- Renamed variables to better reflect the data they actually contain
- Disabled parallelism in test case for better debugging
- Working toward isolating executions on individual files to ease implementing a parallel version in Galaxy
- Added documentation to all extracted functions
- Added stubs for further simplifications which might be more drastical code changes --> those need to be validated in extended tests

Closes #74

### CODE REVIEW

**Comment by wverastegui** [comment]
find.turn.pint.R was reverted to refactored version, Turnpoint function prevents passing the tests. Changelog file was updated accordingly (96c64aed9afc5aa6bbe68b4e8e3f03829548da1e)

File: conda/meta.yaml [hunk 1]
```diff
@@ -62,6 +62,7 @@ requirements:
     - r-tidyr
     - r-stringr
     - r-tibble
+    - r-pastecs
```
Comment (hechth) [inline conda/meta.yaml]: Please remove this from meta.yaml

File: conda/environment-dev.yaml [hunk 2]
```diff
@@ -33,3 +33,4 @@ dependencies:
   - r-datacomparer
   - r-patrick
   - radian
+  - r-pastecs
```
Comment (hechth) [inline conda/environment-dev.yaml]:
```suggestion
```

File: conda/meta.yaml (hunk 1 above)
Comment (hechth) [inline conda/meta.yaml]:
```suggestion
```

**Review by xtrojak — CHANGES_REQUESTED** [review]
Consider spending more time with the merge, I think the final version could benefit from both your and my refactored versions.

File: DESCRIPTION [hunk 3]
```diff
@@ -8,13 +8,13 @@ Authors@R: c(
 Description: This is a customized fork of the original work from Tianwei Yu.
         It takes the adaptive processing of LC/MS metabolomics data further
         with focus on high resolution MS for both LC and GC applications.
-Depends: R (>= 3.50), MASS, mzR, splines, doParallel, foreach, snow, dplyr, 
-        tidyr, stringr, tibble, tools, arrow
+Depends: R (>= 3.50), MASS, rgl, mzR, splines, doParallel, foreach,
+        iterators, snow, gbm, e1071, randomForest, ROCR, ROCS, Rcpp, dplyr, tidyr, stringr, tibble, pastecs, tools, arrow
```
Comment (hechth) [inline DESCRIPTION]:
```suggestion
        iterators, snow, gbm, e1071, randomForest, ROCR, ROCS, Rcpp, dplyr, tidyr, stringr, tibble, tools, arrow
```

**Comment by hechth** [comment]
> Consider spending more time with the merge, I think the final version could benefit from both your and my refactored versions.

Thanks for the hint - I'll have a better look and see how to merge it better! Thanks for paying attention to details :)

File: R/find.tol.time.R [hunk 4]
```diff
@@ -1,24 +1,171 @@
...
+    # exp.y <- this.l$coef[1] + this.l$coef[2] * x
+    # y2 <- y[1:(length(y) - 1)]
+    # y3 <- y[2:(length(y))]
+    # y2[which(y2 < y3)] <- y3[which(y2 < y3)]
+    # y[1:(length(y) - 1)] <- y2
+
+    # yy <- cumsum(y > 1.5 * exp.y)
+    # yi <- seq_along(yy)
+    # sel <- min(which(yy < yi)) - 1
+    # rt_tol_relative <- x[sel]
```
Comment (xtrojak) [inline R/find.tol.time.R]:
```suggestion
```

**Review by xtrojak — APPROVED** [review]

**Pull request merged**